### PR TITLE
Majority of refactoring to integrate ct-core-api.

### PR DESCRIPTION
**Pull request by jcrafford**
Still need to implement todo_resources and test.

### CODE REVIEW

File: flask/requirements.txt [hunk 1]
```diff
@@ -1,12 +1,4 @@
-Flask==0.11.1
```
Comment (bmorgan21) [inline flask/requirements.txt:1]: shouldn't we still be pinning these versions?

File: flask/todo_app/README.rst [hunk 2]
```diff
@@ -0,0 +1,31 @@
+Flask Todo App
+##############
+
+Quick Start
+===========
+
+Set environment vars for the Flask application::
```
Comment (bmorgan21) [inline flask/todo_app/README.rst]: given we are in a docker container, shouldn't this speak to that?

File: flask/todo_app/extensions/principal_ext.py [hunk 3]
```diff
@@ -0,0 +1,8 @@
+from flask_principal import Principal
```
Comment (bmorgan21) [inline flask/todo_app/extensions/principal_ext.py]: I really don't like having the directory information duplicated in the filename.  There has to be a better way to solve the problem.

File: flask/todo_app/models/user_models.py [hunk 4]
```diff
@@ -1,10 +1,12 @@
...
+from flask.ext.login import UserMixin
+
 from todo_app import enums
-from todo_app.models import db
 
 __all__ = ['User', 'Role']
 
 
-class User(db.Model):
+class User(db.Model, UserMixin):
```
Comment (bmorgan21) [inline flask/todo_app/models/user_models.py]: awesome!  cleans up the model!

File: flask/todo_app/models/user_models.py [hunk 5]
```diff
@@ -47,7 +28,4 @@ class Role(db.Model):
 
     @classmethod
     def get_all_for_user_id(cls, user_id):
```
Comment (bmorgan21) [inline flask/todo_app/models/user_models.py]: all these classmethods that are doing queries in models should be removed.  I can clean that up after we get it merged.

File: flask/todo_app/modules/todo/todo_views.py [hunk 6]
```diff
@@ -1,10 +1,13 @@
...
+from ct_core_api.common import app_utils
+
+from flask import Blueprint
 from flask_login import login_required
 
 bp = Blueprint('view.todo', __name__)
 
 
 @bp.route("/")
+@app_utils.template()
```
Comment (bmorgan21) [inline flask/todo_app/modules/todo/todo_views.py]: what does this do?  does it default to index.html for the template by default?

File: flask/todo_app/services/email_service.py [hunk 7]
```diff
@@ -1,16 +1,12 @@
...
-DEFAULT_SENDER = 'Team Todo <help@todo.com>'
+DEFAULT_SENDER = 'Team Todo <help@example.com>'
 
 
 def send_reset_password(user, password):
-    msg = Message("Reset Your Password",
-                  sender=DEFAULT_SENDER,
-                  recipients=[user.email])
-
+    msg = Message("Reset Your Password", sender=DEFAULT_SENDER, recipients=[user.email])
```
Comment (bmorgan21) [inline flask/todo_app/services/email_service.py]: what is your rule for single line vs multiple line?  sometimes I agree w/ you and others I am not sure.

File: flask/todo_app/services/todo_service.py [hunk 8]
```diff
@@ -0,0 +1,13 @@
+from todo_app.lib.model_manager import SqlAlchemyModelManager
+from todo_app.models.todo_models import Todo
+from todo_app.services import Service
+
+__all__ = ['TodoService']
+
+
+class TodoManager(SqlAlchemyModelManager):
```
Comment (bmorgan21) [inline flask/todo_app/services/todo_service.py]: I think managers should be separate.  It is possible that multiple services will leverage the same manager.  i.e. to handle service composition or just things that require transactional integrity.

**Review by bmorgan21 — COMMENTED** [review]
This stuff looks great!  a couple of comments for us to talk through.  Can I actually run it or are there other things to still handle?  i.e. are the ct-\* python libs public?

File: flask/requirements.txt (hunk 1 above)
Comment (jcrafford) [reply flask/requirements.txt:1]: Still pinned, the ones that were removed come from `ct-core-api`

File: flask/todo_app/README.rst (hunk 2 above)
Comment (jcrafford) [reply flask/todo_app/README.rst]: These instructions are for a local development setup that just uses Python and Flask.
Might be helpful on it's own but not as relevant with the code mounted in a docker container. Open to adding more context or removing.

File: flask/todo_app/extensions/principal_ext.py (hunk 3 above)
Comment (jcrafford) [reply flask/todo_app/extensions/principal_ext.py]: We touched on this in chat. Let me know if you still want to see changes.

File: flask/todo_app/modules/todo/todo_views.py (hunk 6 above)
Comment (jcrafford) [reply flask/todo_app/modules/todo/todo_views.py]: It renders the template with the name of the function + ".html" by default.
If the function returns a dict, it'll be used as the template vars.
Just a convenience decorator.

**Pull request merged**